src/IRac.cpp

```diff
@@ -342,6 +346,29 @@ void IRac::coolix(IRCoolixAC *ac,
 }
 #endif  // SEND_COOLIX
 
+#if SEND_CORONA_AC
+void IRac::corona(IRCoronaAc *ac,
+                  const bool on, const stdAc::opmode_t mode,
+                  const float degrees, const stdAc::fanspeed_t fan,
+                  const stdAc::swingv_t swingv, const bool econo) {
+  ac->begin();
+  ac->setPower(on);
```

Comment:
Does PowerToggle need to be set at all?

Reply:
It should probably be included and set yes ;/

Reply:
Best way to think of a `protocol()` method in the `IRac` class is to be fairly dumb.
If a setting is in reality controlled by `setPowerToggle()` or `setSwingVToggle()` etc, then just assume a non-`off` value means Toggle the damn thing.
Try to put the "smarts" in `handleToggles()`
e.g.
It will look at the new and old state and say:
Gosh, old was power off, new is power off. Create & Send an "off" to power as nothing changed. An off in reality when sent to a "toggle" should do nothing.
Similarly, if old was power on, and new is power on. Create & Send an "off" to power as nothing changed.
However, if old was off, and new is on. Then, yes you need to create and send a "on" toggle message. i.e. It changed, so needs to be toggled.

Thus `handleToggles()` tends to XOR the prev and new states together, and create an ephemeral state (that it sends to `IRac::protocol()`) toggle that setting if needed.

Reply:
I think mostly it is the function name that is of, it should already be done in: https://github.com/crankyoldgit/IRremoteESP8266/pull/1152/files#diff-e42bc203289f36e162e6ec65996a0da0R1401-R1411

I will try to work out the power parts better first before going down the `Toggle hole` for that as well.